src/types/observablearray.ts

```diff
@@ -124,8 +133,14 @@ class ObservableArrayAdministration<T> implements IInterceptable<IArrayWillChang
 		if (oldLength !== this.lastKnownLength)
 			throw new Error("[mobx] Modification exception: the internal structure of an observable array was changed. Did you use peek() to change it?");
 		this.lastKnownLength += delta;
-		if (delta > 0 && oldLength + delta > OBSERVABLE_ARRAY_BUFFER_SIZE)
-			reserveArrayBuffer(oldLength + delta);
+		if (delta > 0 && oldLength + delta + 1 > OBSERVABLE_ARRAY_BUFFER_SIZE)
+			reserveArrayBuffer(oldLength + delta + 1);
+		if (delta > 0)
```

Comment:
Ah that works, see https://jsfiddle.net/yfkkk42r/8/ / https://github.com/mobxjs/mobx/commit/0ac2d21a2f9778aac4cf1180a099b2c4fbf49555.

Looking good I think?